qemu/tests/qemu_guest_agent.py

```diff
@@ -4029,6 +4029,50 @@ def get_blocks():
             msg += "the blocks after fstrim is %s." % blocks_after_fstrim
             test.fail(msg)
 
+    @error_context.context_aware
+    def gagent_check_resource_leak(self, test, params, env):
+        """
+        Check ether commands 'guest-get-osinfo/devices' cause
+        resource leak.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+
+        def check_handle(pid):
+            """
+            Check the handles of qga via handle.exe
+
+            :params pid: process ID of qga.service
+            :return qga_handles: the handles of qga.service
+            """
+            cmd_get_handle = params["cmd_get_qga_handle"] % pid
+            qga_handles = int(session.cmd(cmd_get_handle
+                                          ).strip().split(":")[1])
+            return qga_handles
+
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Check whether resource leak can be"
+                              " caused via execute commands "
+                              "'get-osinfo/devices' in a loop.",
+                              logging.info)
+        repeats = int(params.get("repeat_times", 1))
+        pid_qga = int(session.cmd(params["cmd_get_qga_pid"]
+                                  ).strip().split()[1])
+        for i in range(repeats):
+            error_context.context("execute 'get-osinfo/devices'"
```

Comment:
`logging.info` is better, and add ` error_context.context` before `for i in range(repeats):`.